src/Apps/Order/Components/ExpressCheckout/Order2ExpressCheckoutUI.tsx

```diff
@@ -0,0 +1,729 @@
+  SkeletonBox,
+  SkeletonText,
+  Spacer,
+  Text,
+} from "@artsy/palette"
+import {
+  ExpressCheckoutElement,
+  useElements,
+  useStripe,
+} from "@stripe/react-stripe-js"
+import type {
+  AvailablePaymentMethods,
+  ClickResolveDetails,
+  ExpressPaymentType,
+  LineItem,
+  ShippingRate,
+  StripeExpressCheckoutElementClickEvent,
+  StripeExpressCheckoutElementConfirmEvent,
+  StripeExpressCheckoutElementOptions,
+  StripeExpressCheckoutElementShippingAddressChangeEvent,
+  StripeExpressCheckoutElementShippingRateChangeEvent,
+} from "@stripe/stripe-js"
+import { useSetFulfillmentOptionMutation } from "Apps/Order/Components/ExpressCheckout/Mutations/useSetFulfillmentOptionMutation"
+import { useSubmitOrderMutation } from "Apps/Order/Components/ExpressCheckout/Mutations/useSubmitOrderMutation"
+import { useUnsetOrderFulfillmentOptionMutation } from "Apps/Order/Components/ExpressCheckout/Mutations/useUnsetOrderFulfillmentOptionMutation"
+import { useUnsetOrderPaymentMethodMutation } from "Apps/Order/Components/ExpressCheckout/Mutations/useUnsetOrderPaymentMethodMutation"
+import { useUpdateOrderMutation } from "Apps/Order/Components/ExpressCheckout/Mutations/useUpdateOrderMutation"
+import { useUpdateOrderShippingAddressMutation } from "Apps/Order/Components/ExpressCheckout/Mutations/useUpdateOrderShippingAddressMutation"
+import {
+  type OrderMutationSuccess,
+  validateAndExtractOrderResponse,
+} from "Apps/Order/Components/ExpressCheckout/Util/mutationHandling"
+import { useOrderTracking } from "Apps/Order/Hooks/useOrderTracking"
+import { preventHardReload } from "Apps/Order/OrderApp"
+import { useCheckoutContext } from "Apps/Order2/Routes/Checkout/Hooks/useCheckoutContext"
+import { RouterLink } from "System/Components/RouterLink"
+import createLogger from "Utils/logger"
+import type {
+  Order2ExpressCheckoutUI_order$data,
+  Order2ExpressCheckoutUI_order$key,
+} from "__generated__/Order2ExpressCheckoutUI_order.graphql"
+import type {
+  FulfillmentOptionInputEnum,
+  useSetFulfillmentOptionMutation$data,
+} from "__generated__/useSetFulfillmentOptionMutation.graphql"
+import type {
+  OrderCreditCardWalletTypeEnum,
+  useUpdateOrderMutation$data,
+} from "__generated__/useUpdateOrderMutation.graphql"
+import { type Dispatch, type SetStateAction, useRef, useState } from "react"
+import { graphql, useFragment } from "react-relay"
+
+interface Order2ExpressCheckoutUIProps {
+  order: Order2ExpressCheckoutUI_order$key
+  setShowSpinner?: Dispatch<SetStateAction<boolean>>
+  isChrome?: boolean
+}
+
+const logger = createLogger("ExpressCheckoutUI")
+
+// This prop has no type definition in stripe
+type HandleCancelCallback = NonNullable<
+  React.ComponentProps<typeof ExpressCheckoutElement>["onCancel"]
+>
+
+export const Order2ExpressCheckoutUI = ({
+  order,
+  setShowSpinner,
+  isChrome,
+}: Order2ExpressCheckoutUIProps) => {
+  const orderData = useFragment(ORDER_FRAGMENT, order)
+  const [visible, setVisible] = useState(false)
```

Comment:
This might not be necessary since the checkout context loading state will hide this from view (edit: there is some time after loading where the express checkout doesn't appear yet. i dont know if this helps or hurts that. see screenshot from main review comment)